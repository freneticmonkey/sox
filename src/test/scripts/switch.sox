

var number_val = 1;

// testing expression switch value
switch number_val + 1 {
    case 0 {
        print "case 0. Error"
    }
    case 1 {
        print "number_val 1. Error"
    }
    default {
        print "default. Error"
    }
    case 2 {
        print "number_val + 1. Success!"
    }
}

// Testing default
switch number_val + 3 {
    case 0 {
        print "case 0. Error!"
    }
    case 1 {
        print "number_val 1. Error"
    }
    default {
        print "default. Success"
    }
    case 2 {
        print "number_val + 1. Error"
    }
}


// Testing default at end
switch number_val + 3 {
    case 0 {
        print "case 0. Error!"
    }
    case 1 {
        print "number_val 1. Error"
    }
    default {
        print "default. Success"
    }
}

// Testing fallthrough switch
print "Testing fallthrough switch"
switch 0 {
    case 0
    case 1 {
        print "number_val 1 or 0"
    }
    default {
        print "default. Success"
    }
}