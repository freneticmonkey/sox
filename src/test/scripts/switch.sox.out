number_val + 1. Success!
default. Success
default. Success
Testing fallthrough switch
number_val 1 or 0
